core/amber/src/main/python/core/storage/runnables/port_result_writer.py

```diff
@@ -0,0 +1,35 @@
+import queue
+import threading
+
+from core.models import Tuple
+from core.storage.model.buffered_item_writer import BufferedItemWriter
+from core.util import Stoppable
+from core.util.runnable.runnable import Runnable
+
+
+class PortResultWriter(Runnable, Stoppable):
+    def __init__(self, writer: BufferedItemWriter):
+        self.writer: BufferedItemWriter = writer
+        self.queue = queue.Queue()
+        self.stopped = False
+        self.stop_event = threading.Event()
+
+    def put_tuple(self, amber_tuple: Tuple):
+        assert not self.stopped, "Cannot put tuple after termination."
+        self.queue.put(amber_tuple)
```

Comment:
instead of expose `put_tuple` as a method out, I suggest you expose a queue out. and let the threads communicates through queue directly, that's more explicit. Also, you can give a special element into the queue to stop the runnable (that's why it might be better to use `StoppableQueueBlockingRunnable `), rather than using a stop event (this is an additional mutex).

Reply:
Done